crosscluster/physical: TestAlterTenantCompleteToLatest failed

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/2526618f-035b-41a4-b36b-6321778c1b6f?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [e0e47286b08aadf23058825c7b339f19ce852b71](https://github.com/cockroachdb/cockroach/commits/e0e47286b08aadf23058825c7b339f19ce852b71):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest88511943
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:611: waited for 2.730642776s to advance to 1741768088.785652191,0
    alter_replication_job_test.go:87: error executing query="ALTER TENANT $1 START REPLICATION OF $2 ON $3" args=["destination" "source" "postgres://root@127.0.0.1:44099?sslcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest3200809015%2Fcert&sslkey=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest3200809015%2Fkey&sslmode=verify-full&sslrootcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest3200809015%2Fca"]: pq: cannot resume replication into tenant "destination" by reverting to 2025-03-12 08:28:10 +0000 UTC (no history is retained by 0 producer jobs)
    panic.go:629: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest88511943
--- FAIL: TestAlterTenantCompleteToLatest (15.26s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48484

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/a9cc9fda-621d-47d9-9789-5c7ededce788?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [e0e47286b08aadf23058825c7b339f19ce852b71](https://github.com/cockroachdb/cockroach/commits/e0e47286b08aadf23058825c7b339f19ce852b71):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest3354780487
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:611: waited for 6.401517319s to advance to 1741778950.957272448,0
    alter_replication_job_test.go:87: error executing query="ALTER TENANT $1 START REPLICATION OF $2 ON $3" args=["destination" "source" "postgres://root@127.0.0.1:38917?sslcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest2694648367%2Fcert&sslkey=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest2694648367%2Fkey&sslmode=verify-full&sslrootcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest2694648367%2Fca"]: pq: cannot resume replication into tenant "destination" by reverting to 2025-03-12 11:29:15 +0000 UTC (no history is retained by 0 producer jobs)
    panic.go:629: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest3354780487
--- FAIL: TestAlterTenantCompleteToLatest (20.46s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/3362e3b6-e2ca-484b-a8d3-3c847f335c90?testReportRun=3&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [540a0423c37e471e51012b325ec822c03ccb1067](https://github.com/cockroachdb/cockroach/commits/540a0423c37e471e51012b325ec822c03ccb1067):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest4031796060
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:611: waited for 3.361648354s to advance to 1741854427.620968941,0
    alter_replication_job_test.go:87: error executing query="ALTER TENANT $1 START REPLICATION OF $2 ON $3" args=["destination" "source" "postgres://root@127.0.0.1:41999?sslcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F1db3742907b248aaad7daa117b0687cd%2FTestAlterTenantCompleteToLatest2371065337%2Fcert&sslkey=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F1db3742907b248aaad7daa117b0687cd%2FTestAlterTenantCompleteToLatest2371065337%2Fkey&sslmode=verify-full&sslrootcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F1db3742907b248aaad7daa117b0687cd%2FTestAlterTenantCompleteToLatest2371065337%2Fca"]: pq: cannot resume replication into tenant "destination" by reverting to 2025-03-13 08:27:10 +0000 UTC (no history is retained by 0 producer jobs)
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest4031796060
--- FAIL: TestAlterTenantCompleteToLatest (20.47s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/e6e361e5-57e0-48da-b9d8-3ab11474ea0a?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [540a0423c37e471e51012b325ec822c03ccb1067](https://github.com/cockroachdb/cockroach/commits/540a0423c37e471e51012b325ec822c03ccb1067):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest2316419792
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:611: waited for 4.78048618s to advance to 1741863015.914851644,0
    alter_replication_job_test.go:87: error executing query="ALTER TENANT $1 START REPLICATION OF $2 ON $3" args=["destination" "source" "postgres://root@127.0.0.1:38135?sslcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F4b12eaa57b774417f0bd736f4fe28629%2FTestAlterTenantCompleteToLatest3711390354%2Fcert&sslkey=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F4b12eaa57b774417f0bd736f4fe28629%2FTestAlterTenantCompleteToLatest3711390354%2Fkey&sslmode=verify-full&sslrootcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F4b12eaa57b774417f0bd736f4fe28629%2FTestAlterTenantCompleteToLatest3711390354%2Fca"]: pq: cannot resume replication into tenant "destination" by reverting to 2025-03-13 10:50:20 +0000 UTC (no history is retained by 0 producer jobs)
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest2316419792
--- FAIL: TestAlterTenantCompleteToLatest (18.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/78db10ee-46ba-49c9-9721-3d085835ad4c?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [a064b119c4e9e5629cec5c787f09dea29b60d511](https://github.com/cockroachdb/cockroach/commits/a064b119c4e9e5629cec5c787f09dea29b60d511):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest684362368
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:611: waited for 3.162968002s to advance to 1741939619.234600909,0
    alter_replication_job_test.go:87: error executing query="ALTER TENANT $1 START REPLICATION OF $2 ON $3" args=["destination" "source" "postgres://root@127.0.0.1:32793?sslcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F4b12eaa57b774417f0bd736f4fe28629%2FTestAlterTenantCompleteToLatest2235756922%2Fcert&sslkey=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F4b12eaa57b774417f0bd736f4fe28629%2FTestAlterTenantCompleteToLatest2235756922%2Fkey&sslmode=verify-full&sslrootcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F4b12eaa57b774417f0bd736f4fe28629%2FTestAlterTenantCompleteToLatest2235756922%2Fca"]: pq: cannot resume replication into tenant "destination" by reverting to 2025-03-14 08:07:00 +0000 UTC (no history is retained by 0 producer jobs)
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest684362368
--- FAIL: TestAlterTenantCompleteToLatest (16.66s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/934c80a4-ad22-435b-af5a-14b64f27a98a?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [a064b119c4e9e5629cec5c787f09dea29b60d511](https://github.com/cockroachdb/cockroach/commits/a064b119c4e9e5629cec5c787f09dea29b60d511):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest43556735
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:611: waited for 4.430595172s to advance to 1741950346.548780971,0
    alter_replication_job_test.go:87: error executing query="ALTER TENANT $1 START REPLICATION OF $2 ON $3" args=["destination" "source" "postgres://root@127.0.0.1:46489?sslcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest4000847014%2Fcert&sslkey=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest4000847014%2Fkey&sslmode=verify-full&sslrootcert=%2Fvar%2Flib%2Fengflow%2Fworker%2Fwork%2F0%2Fexec%2F_tmp%2F6bd1cd5a50158120b8c1bb763ed421f9%2FTestAlterTenantCompleteToLatest4000847014%2Fca"]: pq: cannot resume replication into tenant "destination" by reverting to 2025-03-14 11:05:50 +0000 UTC (no history is retained by 0 producer jobs)
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest43556735
--- FAIL: TestAlterTenantCompleteToLatest (19.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/crosscluster/physical/alter_replication_job_test.go
+++ b/pkg/crosscluster/physical/alter_replication_job_test.go
@@ -69,7 +69,11 @@ func TestAlterTenantCompleteToLatest(t *testing.T) {
 	c.WaitUntilReplicatedTime(targetReplicatedTime, jobspb.JobID(ingestionJobID))
 
 	var emptyCutoverTime time.Time
-	cutoverOutput := c.Cutover(ctx, producerJobID, ingestionJobID, emptyCutoverTime, false)
+	// Set Async validation to true, so the post cutover retention job has a long
+	// expiration time.
+	cutoverOutput := c.Cutover(ctx, producerJobID, ingestionJobID, emptyCutoverTime, true)
+	jobutils.WaitForJobToSucceed(c.T, c.DestSysSQL, jobspb.JobID(ingestionJobID))
+
 	require.GreaterOrEqual(t, cutoverOutput.GoTime(), targetReplicatedTime.GoTime())
 	require.LessOrEqual(t, cutoverOutput.GoTime(), c.SrcCluster.Server(0).Clock().Now().GoTime())
 	jobutils.WaitForJobToSucceed(c.T, c.DestSysSQL, jobspb.JobID(ingestionJobID))

